alpha/apps/kaltura/lib/live/kBroadcastUrlManager.php

```diff
@@ -211,4 +211,17 @@ public function getSecondaryBroadcastUrl(LiveStreamEntry $entry, $protocol)
 			return $this->getBroadcastUrl($entry, $protocol, $hostname, kBroadcastUrlManager::SECONDARY_MEDIA_SERVER_INDEX, $concatStreamName);
 		}
 	}
+
+	public function getRTCBroadcastingUrl($protocol, $hostname, $externalPort)
+	{
+		if (!$hostname)
+		{
+			return '';
+		}
+		$url = "$protocol://$hostname";
+		if ($externalPort)
```

Comment:
1.i think you ought to rename the $externalport, since we probably need to pass entire URI (for loadbalancers and such)